packages/expo-camera/ios/CameraViewNextModule.swift

```diff
@@ -0,0 +1,247 @@
+          CameraOnlyPermissionRequester(),
+          CameraMicrophonePermissionRequester()
+        ],
+        withPermissionsManager: permissionsManager
+      )
+    }
+
+    Property("modernBarcodeScannerAvailable")
+      .get { () -> Bool in
+        if #available(iOS 16.0, *) {
+          return true
+        }
+        return false
+      }
+
+    // swiftlint:disable:next closure_body_length
+    View(CameraViewNext.self) {
+      Events(cameraEvents)
+
+      Prop("type") { (view, type: CameraTypeNext) in
+        if view.presetCamera != type.toPosition() {
+          view.presetCamera = type.toPosition()
+        }
+      }
+
+      Prop("flashMode") { (view, flashMode: CameraFlashModeNext) in
+        if view.flashMode != flashMode {
+          view.flashMode = flashMode
+        }
+      }
+
+      Prop("enableTorch") { (view, enabled: Bool) in
+        view.torchEnabled = enabled
+      }
+
+      Prop("zoom") { (view, zoom: Double) in
+        if fabs(view.zoom - zoom) > Double.ulpOfOne {
+          view.zoom = zoom
+        }
+      }
+
+      Prop("mode") { (view, mode: CameraModeNext) in
+        if view.mode != mode {
+          view.mode = mode
+        }
+      }
+
+      Prop("barCodeScannerEnabled") { (view, scanBarCodes: Bool?) in
+        if view.isScanningBarCodes != scanBarCodes {
+          view.isScanningBarCodes = scanBarCodes ?? false
+        }
+      }
+
+      Prop("barCodeScannerSettings") { (view, settings: BarcodeSettings) in
+        view.setBarCodeScannerSettings(settings: settings)
+      }
+
+      Prop("mute") { (view, muted: Bool) in
+        view.isMuted = muted
+      }
+
+      Prop("responsiveOrientationWhenOrientationLocked") { (view, responsiveOrientation: Bool) in
+        if view.responsiveWhenOrientationLocked != responsiveOrientation {
+          view.responsiveWhenOrientationLocked = responsiveOrientation
+        }
+      }
+
+      AsyncFunction("takePicture") { (view, options: TakePictureOptionsNext, promise: Promise) in
+        #if targetEnvironment(simulator)
+        try takePictureForSimulator(self.appContext, view, options, promise)
+        #else // simulator
+        view.takePicture(options: options, promise: promise)
+        #endif // not simulator
+      }.runOnQueue(.main)
+
+      AsyncFunction("record") { (view, options: CameraRecordingOptionsNext, promise: Promise) in
+        #if targetEnvironment(simulator)
+        throw Exceptions.SimulatorNotSupported()
+        #else
+        view.record(options: options, promise: promise)
+        #endif
+      }.runOnQueue(.main)
+
+      AsyncFunction("stopRecording") { view in
+        #if targetEnvironment(simulator)
+        throw Exceptions.SimulatorNotSupported()
+        #else
+        view.stopRecording()
+        #endif
+      }.runOnQueue(.main)
+    }
+
+    AsyncFunction("launchModernScanner") { (options: VisionScannerOptions?, promise: Promise) in
+      if #available(iOS 16.0, *) {
```

Comment:
Why is this if needed? If you don't put it here, you could remove `Task {}` because `AsyncFunction` does support async/await functions (don't forget to remove the promise)

Reply:
`VisionScannerDelegate` is marked as `iOS 16`, so I can't use it without it,